.gitignore

```diff
@@ -1,4 +1,17 @@
+# Binary files
 /app
```

Comment:
`/app` is carried over from other repos' `.gitignore`s since it's the folder that the default Make target usually builds the binary to. This repo doesn't have a default Make target that builds the binary so I can remove this.

`path-auth-data-server` is the filename of the built binary if a user were to run `go build .` from the root so it's just there as a failsafe and I think makes sense to keep.